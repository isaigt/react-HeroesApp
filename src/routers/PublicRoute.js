import React from 'react';
import PropTypes from 'prop-types';
import { Redirect, Route } from 'react-router-dom';

export const PublicRoute = ({
    isAuthenticated,
    component: Component,
    ...rest
}) => {
    const lastPath = localStorage.getItem('lastPath') || '/';

    return (
        <Route
            { ...rest }
            component={
                ( props ) => (
                    ( !isAuthenticated )
                    ? <Component { ...props } />
                    : <Redirect to={ lastPath } />
                )
            }
        />
    )
}

PublicRoute.propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired
}