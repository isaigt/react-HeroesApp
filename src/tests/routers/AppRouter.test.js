import { mount } from 'enzyme';
import { AuthContext } from '../../auth/authContext';
import { AppRouter } from '../../routers/AppRouter';

describe('Pruebas del componente <AppRouter />', () => {

    const context = {
        dispatch: jest.fn(),
        user: {
            name: 'Isai',
            logged: false
        }
    }
    
    test('debe de mostrar el login si no esta autenticado', () => {
        const wrapper = mount(
            <AuthContext.Provider value={ context } >
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar el componente marvel si esta autenticado', () => {
        const context = {
            dispatch: jest.fn(),
            user: {
                name: 'Isai',
                logged: true
            }
        }
        const wrapper = mount(
            <AuthContext.Provider value={ context } >
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper.find('.navbar').exists() ).toBe( true );
    });
});
