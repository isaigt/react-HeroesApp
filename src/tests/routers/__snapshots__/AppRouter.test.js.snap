// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <AppRouter /> debe de mostrar el login si no esta autenticado 1`] = `
<div>
  <div
    className="container mt-5 text-center"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Iniciar Sesion
    </button>
  </div>
</div>
`;
